tools/swift-inspect/Sources/SwiftInspectLinux/ElfPhdr.swift

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+internal protocol ElfPhdr {
+  static var symbolSize: Int { get }
+  var type: UInt32 { get }
+  var offset: UInt64 { get }
+  var vaddr: UInt64 { get }
+  var paddr: UInt64 { get }
+  var filesz: UInt64 { get }
+  var memsz: UInt64 { get }
+  var flags: UInt32 { get }
+  var align: UInt64 { get }
+}
+
+extension Elf64_Phdr: ElfPhdr {
+  static var symbolSize: Int { MemoryLayout<Elf64_Phdr>.size }
+  var type: UInt32 { self.p_type }
+  var offset: UInt64 { self.p_offset }
+  var vaddr: UInt64 { self.p_vaddr }
+  var paddr: UInt64 { self.p_paddr }
+  var filesz: UInt64 { self.p_filesz }
+  var memsz: UInt64 { self.p_memsz }
+  var flags: UInt32 { self.p_flags }
+  var align: UInt64 { self.p_align }
+}
+
+extension Elf32_Phdr: ElfPhdr {
+  static var symbolSize: Int { MemoryLayout<Elf32_Phdr>.size }
+  var type: UInt32 { self.p_type }
+  var offset: UInt64 { UInt64(self.p_offset) }
+  var vaddr: UInt64 { UInt64(self.p_vaddr) }
+  var paddr: UInt64 { UInt64(self.p_paddr) }
+  var filesz: UInt64 { UInt64(self.p_filesz) }
+  var memsz: UInt64 { UInt64(self.p_memsz) }
+  var flags: UInt32 { self.p_flags }
+  var align: UInt64 { UInt64(self.p_align) }
+}
```

Comment:
Please define these in a C header

Reply:
Can I move this to a C header? It is an extension to the existing `Elf32_Phdr` C struct which is imported via `elf.h`. My goal with this file (and the other `Elf*.swift` files) was to create a generic way of accessing both elf32 and elf64 structs, and this protocol + extending of the C structs pattern is what I came up with. Is there a better way to achieve the same?

Reply:
This looks good, I think @compnerd misread this as defining the type rather than just extending it.